Source/Core/DolphinQt/Debugger/CodeDiffDialog.cpp

```diff
@@ -133,6 +147,48 @@ void CodeDiffDialog::OnClickItem()
   m_code_widget->SetAddress(address, CodeViewWidget::SetAddressUpdate::WithDetailedUpdate);
 }
 
+void CodeDiffDialog::SaveDataBackup()
+{
+  std::string filename =
+      File::GetUserPath(D_LOGS_IDX) + SConfig::GetInstance().GetGameID() + "_CodeDiff.txt";
+  File::IOFile f(filename, "w");
+  if (!f)
+  {
+    PanicAlertFmt("Failed to open {}", filename);
+    return;
+  }
+
+  for (const auto& line : m_include)
+    f.WriteString(fmt::format("{} {} {} {}\n", line.addr, line.hits, line.total_hits, line.symbol));
+}
+
+void CodeDiffDialog::LoadDataBackup()
+{
+  std::string filename =
+      File::GetUserPath(D_LOGS_IDX) + SConfig::GetInstance().GetGameID() + "_CodeDiff.txt";
+  File::IOFile f(filename, "r");
+  if (!f)
+  {
+    PanicAlertFmt("Failed to open {}", filename);
+    return;
+  };
+
+  ClearData();
+
+  char line[512];
+  while (fgets(line, 512, f.GetHandle()))

```

Comment:
I may be overthinking/missing something, but couldn't this lead to a buffer overflow?
Consider if I have a symbol name of an absurd length, would it cause any issues?

Reply:
iirc I copied how symbols are loaded. We lose some chars for the other data though. Maybe I can check if there's a max symbol length.

Reply:
If it's larger than the buffer, it just gets chopped off. The symbol map loading also has extra address data in it, so it actually has a lower maximum symbol length than this save/load call.